### PR TITLE
Woo-Connect Insights: Add Top Sellers to Orders Page

### PR DESCRIPTION
**Pull request by psealock**
### Render Top Products
screen shot 2017-06-02 at 4 59 20 pm

Fixes https://github.com/Automattic/wp-calypso/issues/14499
Fixes https://github.com/Automattic/wp-calypso/issues/14498

### CODE REVIEW

File: client/extensions/woocommerce/app/store-stats/index.js [hunk 1]
```diff
@@ -30,6 +31,23 @@ class StoreStats extends Component {
...
+		const translate = this.props.translate;
+		const moduleStrings = {
+			title: translate( 'Products', { context: 'Stats: title of module' } ),
+			item: translate( 'Title', { context: 'Stats: module row header for product title.' } ),
+			value: translate( 'Qty', { context: 'Stats: module row header for number of products sold.' } ),
+			values: [
+				{ value: translate( 'Qty', { context: 'Stats: module row header for number of products sold.' } ) },
+				{ value: translate( 'Total', { context: 'Stats: module row header for total amount sold.' } ) },
+			],
+			empty: translate( 'No products found', { context: 'Stats: Info box label when the Products module is empty' } )
```
Comment (a8ci18n) [inline client/extensions/woocommerce/app/store-stats/index.js]: :information_source: This **context** is really long. Are you sure you don't want to use a translator comment instead?

File: client/extensions/woocommerce/app/store-stats/index.js [hunk 2]
```diff
@@ -30,6 +31,23 @@ class StoreStats extends Component {
...
+			limit: '3',
+		};
+		const translate = this.props.translate;
+		const moduleStrings = {
+			title: translate( 'Products', { context: 'Stats: title of module' } ),
+			item: translate( 'Title', { context: 'Stats: module row header for product title.' } ),
+			value: translate( 'Qty', { context: 'Stats: module row header for number of products sold.' } ),
+			values: [
+				{ value: translate( 'Qty', { context: 'Stats: module row header for number of products sold.' } ) },
+				{ value: translate( 'Total', { context: 'Stats: module row header for total amount sold.' } ) },
```
Comment (a8ci18n) [inline client/extensions/woocommerce/app/store-stats/index.js]: Hi! I've found a possible matching string that has already been translated **38** times:
`translate( 'Total:', { context: 'Upgrades: Total cart cost in checkout when buying a full price upgrade'} )` *ES Score: 11*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/woo-stats-topSellers)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

:information_source: This **context** is really long. Are you sure you don't want to use a translator comment instead?

File: client/extensions/woocommerce/app/store-stats/index.js [hunk 3]
```diff
@@ -30,6 +31,23 @@ class StoreStats extends Component {
...
+			unit,
+			date: selectedDate,
+			quantity: '7',
+			limit: '3',
+		};
+		const translate = this.props.translate;
+		const moduleStrings = {
+			title: translate( 'Products', { context: 'Stats: title of module' } ),
+			item: translate( 'Title', { context: 'Stats: module row header for product title.' } ),
+			value: translate( 'Qty', { context: 'Stats: module row header for number of products sold.' } ),
```
Comment (a8ci18n) [inline client/extensions/woocommerce/app/store-stats/index.js]: :information_source: This **context** is really long. Are you sure you don't want to use a translator comment instead?

File: client/extensions/woocommerce/app/store-stats/index.js [hunk 4]
```diff
@@ -30,6 +31,23 @@ class StoreStats extends Component {
...
+		const topSellersQuery = {
+			unit,
+			date: selectedDate,
+			quantity: '7',
+			limit: '3',
+		};
+		const translate = this.props.translate;
+		const moduleStrings = {
+			title: translate( 'Products', { context: 'Stats: title of module' } ),
+			item: translate( 'Title', { context: 'Stats: module row header for product title.' } ),
```
Comment (a8ci18n) [inline client/extensions/woocommerce/app/store-stats/index.js]: Hi! I've found a possible matching string that has already been translated **47** times:
`translate( 'Title', { context: 'Stats: module row header for post title.'} )` *ES Score: 14*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/woo-stats-topSellers)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

:information_source: This **context** is really long. Are you sure you don't want to use a translator comment instead?

File: client/extensions/woocommerce/app/store-stats/index.js [hunk 5]
```diff
@@ -30,6 +31,23 @@ class StoreStats extends Component {
...
 		};
+		const topSellersQuery = {
+			unit,
+			date: selectedDate,
+			quantity: '7',
+			limit: '3',
+		};
+		const translate = this.props.translate;
+		const moduleStrings = {
+			title: translate( 'Products', { context: 'Stats: title of module' } ),
```
Comment (a8ci18n) [inline client/extensions/woocommerce/app/store-stats/index.js]: Hi! I've found a possible matching string that has already been translated **7** times:
`translate( 'Products' )` *ES Score: 14*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/woo-stats-topSellers)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

:information_source: This **context** is really long. Are you sure you don't want to use a translator comment instead?

File: client/extensions/woocommerce/app/store-stats/index.js [hunk 6]
```diff
@@ -30,6 +31,23 @@ class StoreStats extends Component {
...
+			item: translate( 'Title', { context: 'Stats: module row header for product title.' } ),
+			value: translate( 'Total', { context: 'Stats: module row header for total amount sold.' } ),
+			values: [
+				{ quantity: translate( 'Qty', { context: 'Stats: module row header for number of products sold.' } ) },
+				{ total: translate( 'Total', { context: 'Stats: module row header for total amount sold.' } ) },
+			],
+			empty: translate( 'No products found', { context: 'Stats: Info box label when the Products module is empty' } )
```
Comment (a8ci18n) [inline client/extensions/woocommerce/app/store-stats/index.js]: :information_source: This **context** is really long. Are you sure you don't want to use a translator comment instead?

File: client/extensions/woocommerce/app/store-stats/index.js [hunk 7]
```diff
@@ -30,6 +31,23 @@ class StoreStats extends Component {
...
+			unit,
+			date: selectedDate,
+			quantity: '7',
+			limit: '3',
+		};
+		const translate = this.props.translate;
+		const moduleStrings = {
+			title: translate( 'Products', { context: 'Stats: title of module' } ),
+			item: translate( 'Title', { context: 'Stats: module row header for product title.' } ),
+			value: translate( 'Total', { context: 'Stats: module row header for total amount sold.' } ),
```
Comment (a8ci18n) [inline client/extensions/woocommerce/app/store-stats/index.js]: Hi! I've found a possible matching string that has already been translated **38** times:
`translate( 'Total:', { context: 'Upgrades: Total cart cost in checkout when buying a full price upgrade'} )` *ES Score: 11*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/woo-stats-topSellers)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

:information_source: This **context** is really long. Are you sure you don't want to use a translator comment instead?

File: client/extensions/woocommerce/app/store-stats/index.js [hunk 8]
```diff
@@ -30,6 +31,23 @@ class StoreStats extends Component {
...
+			values: [
+				{ quantity: translate( 'Qty', { context: 'Stats: module row header for number of products sold.' } ) },
```
Comment (a8ci18n) [inline client/extensions/woocommerce/app/store-stats/index.js]: :information_source: This **context** is really long. Are you sure you don't want to use a translator comment instead?

File: client/extensions/woocommerce/app/store-stats/index.js [hunk 9]
```diff
@@ -30,6 +31,23 @@ class StoreStats extends Component {
...
+		const translate = this.props.translate;
+		const moduleStrings = {
+			title: translate( 'Products', { context: 'Stats: title of module' } ),
+			item: translate( 'Title', { context: 'Stats: module row header for product title.' } ),
+			value: translate( 'Total', { context: 'Stats: module row header for total amount sold.' } ),
+			values: [
+				{ key: 'total', value: translate( 'Total', { context: 'Stats: module row header for total amount sold.' } ) },
+				{ key: 'quantity', value: translate( 'Qty', { context: 'Stats: module row header for number of products sold.' } ) },
+			],
+			empty: translate( 'No products found', { context: 'Stats: Info box label when the Products module is empty' } )
```
Comment (a8ci18n) [inline client/extensions/woocommerce/app/store-stats/index.js]: :information_source: This **context** is really long. Are you sure you don't want to use a translator comment instead?

File: client/extensions/woocommerce/app/store-stats/index.js [hunk 10]
```diff
@@ -30,6 +31,23 @@ class StoreStats extends Component {
...
+			limit: '3',
+		};
+		const translate = this.props.translate;
+		const moduleStrings = {
+			title: translate( 'Products', { context: 'Stats: title of module' } ),
+			item: translate( 'Title', { context: 'Stats: module row header for product title.' } ),
+			value: translate( 'Total', { context: 'Stats: module row header for total amount sold.' } ),
+			values: [
+				{ key: 'total', value: translate( 'Total', { context: 'Stats: module row header for total amount sold.' } ) },
+				{ key: 'quantity', value: translate( 'Qty', { context: 'Stats: module row header for number of products sold.' } ) },
```
Comment (a8ci18n) [inline client/extensions/woocommerce/app/store-stats/index.js]: :information_source: This **context** is really long. Are you sure you don't want to use a translator comment instead?

**Comment by psealock** [comment]
@blowery I'm not sure how familiar you are with Site Stats, but I attempting to reuse as many components as possible. Could you give this a look? I want to know if I'm headed in the right direction.

### Progress So Far
screen shot 2017-06-07 at 4 29 29 pm

### Problem
The current stats-list, [baked into the stats-module](https://github.com/Automattic/wp-calypso/blob/master/client/my-sites/stats/stats-module/index.jsx#L156), does not allow for multiple columns. We'll need a proper table to suit our needs. Decoupling these components to reuse stats-module involves a not insignificant refactor which I'd like to tackle at a future point. I attempted to reuse stats-list by making it a table with multiple columns. That's no small task either, and one that doesn't necessarily get us further ahead.

### My direction
I'm writing the a store-stats-module to go the direction stats-module should have gone. Its only concerns are to initiate a query for data, display a `loading` state, and provide a message in the event of no data present. If data is present, it renders its children.

The table, `store-stats-list` will also be used for the List View, https://dynamicsp2.files.wordpress.com/2017/05/listview.png. Thats very much a work in progress.

Any input you may have would be great.

File: client/extensions/woocommerce/app/store-stats/index.js [hunk 11]
```diff
@@ -12,6 +12,10 @@ import Main from 'components/main';
 import Navigation from './store-stats-navigation';
 import { getSelectedSiteId } from 'state/ui/selectors';
 import Chart from './store-stats-chart';
+import Module from './store-stats-module';
+import List from './store-stats-list';
+import SectionHeader from 'components/section-header';
+import StoreStrings from 'woocommerce/app/store-stats/store-strings';
```
Comment (greenafrican) [inline client/extensions/woocommerce/app/store-stats/index.js]: Maybe `import { topProducts } from 'woocommerce/app/store-stats/store-strings';`?

File: client/extensions/woocommerce/app/store-stats/index.js [hunk 12]
```diff
@@ -30,6 +34,16 @@ class StoreStats extends Component {
 			date: today,
 			quantity: '30'
 		};
+		const topSellersQuery = {
+			unit,
+			date: selectedDate,
+			quantity: '7',
+			limit: '3',
+		};
+		const { topProducts } = StoreStrings;
```
Comment (greenafrican) [inline client/extensions/woocommerce/app/store-stats/index.js]: Remove

**Comment by greenafrican** [comment]
Looks like a good start @psealock 

I initially thought it would be great if we could use these lists for the widgets too:
screen shot 2017-06-07 at 4 09 10 pm

However, I think we'd need a seperate query for each row and the data structure would be quite different.

Did you see the list component that the folks at WooCommerce in Calypso created in the WooCommerce extension - https://github.com/Automattic/wp-calypso/pull/13956 ?

File: client/extensions/woocommerce/app/store-stats/store-stats-list/index.js [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+} from 'state/stats/lists/selectors';
+import Card from 'components/card';
+
+const StoreStatsList = ( { data, label, values } ) => {
+	return (
+		<Card compact>
+			<table>
+				<thead>
+				<tr>
+					<th>{ label.value }</th>
```
Comment (blowery) [inline client/extensions/woocommerce/app/store-stats/store-stats-list/index.js]: does this one need a `key` too?

File: client/extensions/woocommerce/app/store-stats/index.js [hunk 14]
```diff
@@ -30,6 +34,16 @@ class StoreStats extends Component {
...
+		const topProductsHeader = (
+			<SectionHeader href={ '/store/stats/products' }>{ topProducts.title }</SectionHeader>
```
Comment (blowery) [inline client/extensions/woocommerce/app/store-stats/index.js]: no need for `{}`, can just use `""` since it's a plain string

**Comment by psealock** [comment]
@greenafrican 

>I initially thought it would be great if we could use these lists for the widgets too:

I'm intending to make it as open as possible so we can do this.

**Comment by psealock** [comment]
Tabular data created here https://github.com/Automattic/wp-calypso/pull/14892

Blocked until this^^^ gets merged

**Comment by psealock** [comment]
@greenafrican Ready for review, thanks

**Comment by psealock** [comment]
Holding off until https://github.com/Automattic/wp-calypso/pull/14911 is merged. A lot of shared logic and components

**Comment by greenafrican** [comment]
LGTM but needs a rebase and update to support the updated period formats from https://github.com/Automattic/wp-calypso/pull/15559

**Pull request merged**